CleanWeather/Scenes/CityList/CityListWorker.swift

```diff
@@ -49,4 +57,33 @@ private extension CityListWorkerImpl {
     private func sortCity(city: [CityWeather]) -> [CityWeather] {
         return city.sorted(by: { $0.city < $1.city })
     }
+
+    private func handleCityWeatherResponse(city: City, response: Result<CityListAPIResponse, Error>, completion: FetchWeatherCompletion?) {
```

Comment:
to rozbić bardziej można 
zrób konwerter Result<CityListApiResponsa> na Result<CityWeathera>

przekształcaj tak na bieżąco odpowiedzi i te result trzymaj w array 
przekazując ta modyfikowana za każdym razem listę dalej do kolejnych handleCompletion będziesz mógł bez counterow sprawdzić czy wszystko już jest, responses.count == cities.count

nie rób switch result,sukces,failure za każdym razem, tylko na samym końcu jak już wszystkie odpowiedzi są, lecisz po wszystkich responsach i jak jest error to zwracasz error a jak nie to sukces